core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -1626,6 +1642,63 @@ private ExprValue exprToDays(ExprValue date) {
     return new ExprLongValue(date.dateValue().toEpochDay() + DAYS_0000_TO_1970);
   }
 
+  //Helper function to handle a '0000' year for the 'to_seconds' function according to MySQL.
+  private ExprValue handleZeroYear(ExprValue date) {
+    String dateStringVal = date.stringValue();
+
+    //Handle corner case for date of 0000-00-00 according to MySQL docs.
+    if (dateStringVal.contains("0000-00-00")){
+      return ExprNullValue.of();
+    }
+
+    //1970 used as a placeholder to extract correct seconds value.
+    if(dateStringVal.length() > 10) {
+      return(new ExprLongValue(
+          LocalDateTime.parse(date.stringValue().replace("0000-", "1970-"))
+              .toEpochSecond(ZoneOffset.UTC))
```

Comment:
Maybe: don't worry about parsing a string, it will be properly implemented in #856 for all functions. Remove signature which accepts `STRING` argument. Existing automatic cast `STRING` -> `TIME/DATE/DATETIME/TIMESTAMP` should handle some cases.

Reply:
Removed in https://github.com/Bit-Quill/opensearch-project-sql/commit/9a2377a5b8ee4bcf5df64a5cea5bf20bd069d32b